src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -24,28 +29,39 @@ public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books",
+            description = "Get a list of all existing books")
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get book by ID",
+            description = "Return a book by identifier")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.findById(id);
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create new book",
+            description = "Create a new book")
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @PutMapping("/{id}")
+    @Operation(summary = "Update book by ID",
+            description = "Updates book parameters with a new one")
     public BookDto updateBookById(@PathVariable Long id,
-                                  @RequestBody CreateBookRequestDto requestDto) {
-        return bookService.update(id, requestDto);
+                                  @RequestBody @Valid CreateBookRequestDto requestDto) {
+        return bookService.updateById(id, requestDto);
     }
 
-    @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @Operation(summary = "Delete book by ID",
+            description = "\"Softly\" removes the book")
```

Comment:
I don't think that the user must know that delete is soft. 
It is our implementation and user shouldn't know that, in otherwise, he will start asking questions about how to delete the book forever.